src/core/stdc/stdlib.d

```diff
@@ -159,6 +159,17 @@ else
 ///
 void*   malloc(size_t size) pure;
 ///
+void*   pureMalloc(size_t size) pure
+{
+    const errno = pureGetErrno();
+    typeof(return) ret = malloc(size);
+    cast(void)pureSetErrno(errno);
+    return ret;
+}
+extern (C) pure pragma(mangle, "getErrno") int pureGetErrno(); // for internal use
+extern (C) pure pragma(mangle, "setErrno") int pureSetErrno(int); // for internal use
```

Comment:
> does this work on LDC?

Multiple `extern (C)` forward-declarations with the same mangled name aren't a problem. [This](http://ldc.acomirei.ru/#compilers:!((compiler:ldc,options:'',source:'extern+(C)+int+foo()%3B%0A%0Apragma+(mangle,+%22foo%22)%0Aextern+(C)+int+bar()+pure%3B%0A%0Avoid+main()%0A%7B%0A++foo()%3B%0A++bar()%3B%0A%7D%0A')),filterAsm:(commentOnly:!t,directives:!t,labels:!t),version:3) works.